warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/ErrorShardedIngestHelper.java

```diff
@@ -1,39 +1,284 @@
 package datawave.ingest.data.config.ingest;
 
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Pattern;
+
 import org.apache.hadoop.conf.Configuration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import com.google.common.collect.HashMultimap;
 import com.google.common.collect.Multimap;
+import com.google.common.collect.Sets;
 
 import datawave.ingest.data.RawRecordContainer;
+import datawave.ingest.data.Type;
+import datawave.ingest.data.TypeRegistry;
 import datawave.ingest.data.config.NormalizedContentInterface;
 
 /**
  *
  */
 public class ErrorShardedIngestHelper extends BaseIngestHelper {
 
+    private static final Logger log = LoggerFactory.getLogger(ErrorShardedIngestHelper.class);
+
+    /**
+     * <p>
+     * Configuration inserted between the <datatype> and the <field-constant> for explicit error parsing.
+     * </p>
+     * <p>
+     * For example:
+     * </p>
+     * <p>
+     * {@code String target = <some-datatype> + DATATYPE_ERROR + INDEX_FIELDS;}
+     * </p>
+     */
+    private static final String ERROR = "error";
+    private IngestHelperInterface delegate = null;
+
+    private Map<Type,IndexedFields> errorIndexedFields = new HashMap<>();
+    private Map<Type,IndexedFields> errorReverseIndexedFields = new HashMap<>();
+
+    protected boolean hasErrorIndexDisallowlist = false;
```

Comment:
These properties need to be moved into the `IndexedFields` class and set appropriately. They are datatype-specific too.